Cython/Compiler/PyrexTypes.py

```diff
@@ -5460,16 +5460,30 @@ def cap_length(s, max_prefix=63, max_len=1024):
     hash_prefix = hashlib.sha256(s.encode('ascii')).hexdigest()[:6]
     return '%s__%s__etc' % (hash_prefix, s[:max_len-17])
 
-def write_noexcept_performance_hint(pos, env, function_name=None, void_return=False):
-    on_what = "on '%s' " % function_name if function_name else ""
+def write_noexcept_performance_hint(pos, env, function_name=None, void_return=False, is_call=False):
+    if function_name:
+        if is_call:
+            on_what = "after calling '%s' " % function_name
+        else:
+            on_what = "on '%s' " % function_name
+    else:
+        if is_call:
+            on_what = "after function call"
```

Comment:
```suggestion
            on_what = "after function call "
```
If I see correctly, the `msg` below requires space as last character in `on_what`.